flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -310,6 +310,12 @@ class TaskManager(
 
     case FatalError(message, cause) =>
       killTaskManagerFatal(message, cause)
+
+    case RequestTaskManagerLog(requestType : LogTypeRequest) =>
+      handleRequestTaskManagerLog(sender(), requestType, currentJobManager.get)
```

Comment:
Why not simply returning a `Future[BlobKey]` from the `handleRequestTaskManagerLog` which is simply piped to the `sender`?